generic3g/ComponentSpecParser.F90

```diff
@@ -177,6 +188,61 @@ subroutine to_typekind(typekind, attributes, rc)
          _RETURN(_SUCCESS)
       end subroutine to_typekind
 
+      subroutine to_VerticalDimSpec(vertical_dim_spec, attributes, rc)
+         type(VerticalDimSpec) :: vertical_dim_spec
+         class(YAML_Node), intent(in) :: attributes
+         integer, optional, intent(out) :: rc
+
+         integer :: status
+         character(:), allocatable :: vertical_str
+
+         vertical_dim_spec = VERTICAL_DIM_NONE ! GEOS default
+         
+         if (.not. attributes%has('vertical_dim_spec')) then
+            _RETURN(_SUCCESS)
+         end if
+         call attributes%get(vertical_str, 'vertical_dim_spec', _RC)
+
+         select case (vertical_str)
+         case ('vertical_dim_none')
+            vertical_dim_spec = VERTICAL_DIM_NONE
+         case ('vertical_dim_center')
+            vertical_dim_spec = VERTICAL_DIM_CENTER
+         case ('vertical_dim_edge')
+            vertical_dim_spec = VERTICAL_DIM_EDGE
+         case default
+            _FAIL('Unsupported typekind')
+         end select
```

Comment:
For convenience we might want shorthands for these strings ala current state spec files.  E.g.,
```suggestion
         select case (vertical_str)
         case ('vertical_dim_none', 'N')
            vertical_dim_spec = VERTICAL_DIM_NONE
         case ('vertical_dim_center', 'C')
            vertical_dim_spec = VERTICAL_DIM_CENTER
         case ('vertical_dim_edge', 'E')
            vertical_dim_spec = VERTICAL_DIM_EDGE
         case default
            _FAIL('Unsupported typekind')
         end select
```